docs/quality_assurance.md

```diff
@@ -0,0 +1,157 @@
+
+The verification of Interchain Security is done using the following approaches:
+- code review
+- automatic software engineering tools (e.g., SonarCloud, gosec)
+- unit testing
+- differential testing
+- incentivized testnet
+- protocol audit (with the Apalache team from Informal Systems)
+
+The verification of Interchain Security is split across the following concerns:
+- Correct software engineering practices (e.g., error handling, serialization, deserialization).
+- The correct integration with IBC (i.e., [ibc-go](https://github.com/cosmos/ibc-go/tree/v3.0.0)).
+- The correct integration with Cosmos SDK (i.e., [cosmos-sdk](https://github.com/cosmos/cosmos-sdk/tree/v0.45.6)).
+- The correctness of the provider chain, i.e., the provider CCV module does not break the liveness or the safety of the provider chain. 
+- The correctness of the Interchain Security protocol, i.e., the protocol follows the [specification](https://github.com/cosmos/ibc/blob/master/spec/appics-028-cross-chain-validation/README.md).
+- The correctness of the consumer chain, i.e., both liveness or safety hold.
+
+> TODO decide what should be covered by the protocol audit and the priority
+
+## Software Engineering
+
+| Concern | Code Review | Automatic Tools | Unit Testing |
+| ------- | ----------- | --------------- | ------------ |
+| Unexpected panics | `TODO` | `??` | `??` |
+| Handling errors | `TODO` | `??` | `??` |
+| Accessing state (setters, getters, iterators) | `TODO` | `??` | `Low coverage` |
+| Serialization / deserialization | `TODO` | `??` | `??` |
+
+## Integration with IBC
+
+Interchain Security is an IBC application and, thus, it relies on IBC to establish a separate channel between the provider chain and every consumer chain. Interchain Security relies on the IBC v3.0 golang [implementation](https://github.com/cosmos/ibc-go/tree/v3.0.0).
+
+IBC packets:
+- ValidatorSetChangePacket
+- MaturedVSCPacket
+- SlashPacketData
+
+| Concern | Code Review | Unit Testing | Diff. testing | Testnet | Protocol audit |
+| ------- | ----------- | ------------ | ------------- | ------- | ----- |
+| Create IBC clients | `TODO` (ibc-go team) | `Done` | `??` | `TODO` | `NA` |
+| Getting consumer `UnbondingPeriod` from IBC client | `TODO` (ibc-go team) | `??` | `??` | `TODO` | `NA` |
+| Create CCV channel (handshake) | `TODO` (ibc-go team) | `Done` | `??` | `TODO` | `TODO` |
+| Sending IBC packets <br /> - see `x/ccv/utils/utils.go:SendIBCPacket()` | `TODO` (ibc-go team) | `Done` | `??` | `TODO` | `NA` |
+| Handling acknowledgments | `TODO` (ibc-go team) | `Low coverage` | `??` | `TODO` | `NA` |
+| Handling timeouts | `TODO` (ibc-go team) | `Low coverage` | `??` | `TODO` | `NA` |
+| Handling IBC client expiration | `TODO` (ibc-go team) | `??` | `??` | `TODO` | `NA` |
+| ICS-20 channel creation | `TODO` (ibc-go team) | `??` | `??` | `TODO` | `NA` |
+| ICS-20 transfer | `TODO` (ibc-go team) | `??` | `??` | `TODO` | `NA` |
+
```

Comment:
| Changes in IBC-GO testing suite | `TODO` (ibc-go team) | `??` | `??` | `TODO` | `NA` |